### DISCUSSION TITLE
Unexpected behaviour with user group permissions

### DISCUSSION THREAD

**Discussion by LauraMontgomery** [solspace/craft-freeform#872] [Feature Requests]
Sorry if this is expected behaviour. And if so, how I should go about it instead?

I have a user group which behaves as the Freeform Admin user group - with the idea begin that they can manage what forms other user groups can read, manage, etc. I have selected 'Read All Submissions', 'Manage All Submissions', and 'Manage All Forms', which this group can do. So far so good.

I have a second user group with more restricted access to Freeform. They have 'Read Submissions by Form', 'Manage Submissions by Form', and 'Manage Forms Individually' selected, but none of the individual forms for each selected as default. I understand at this point, they won't see any forms.

The idea is that the Freeform Admin group can then select individual forms for each user in the second group. But unfortunately they can't even see the forms in the second user groups settings. Weirdly however, the Freeform Admin group can give the second group 'Read All Submissions', 'Manage All Submissions', and 'Manage All Forms' access.

In order for the Freeform Admin group to be able to give the second group users access to individual forms, I need to also select 'Read Submissions by Form', 'Manage Submissions by Form', and 'Manage Forms Individually', plus all the individual forms. Only at this point do I see the deselected forms in the second user group.

This is a problem because the client creates a lot of new forms for registering for events, etc. And therefore the only way for the Freeform Admin group to be able to assign the second group access to the new forms is for Craft Admins to update the Freeform Admin permissions to each new form.

I would have expected the 'Read All Submissions', 'Manage All Submissions', and 'Manage All Forms' permissions to enable the individual user settings on other groups, rather than also needing to select the individual options and forms to achieve this. But perhaps this isn't how this is intended to work? Or perhaps I am missing another setting somewhere?

- Craft Version: 4.5.5
- Freeform Version: 4.1.8
- Freeform Edition: Pro

**Comment by kjmartens** [comment]
Hi @LauraMontgomery,

This doesn't seem right, and I am able to duplicate the behavior you're experiencing. I've noted it and we'll have a developer check into it soon. 🙂

**Comment by LauraMontgomery** [comment]
Thanks @kjmartens 👍

**Comment by kjmartens** [comment]
Hi @LauraMontgomery,

We spent some time investigating this further, and it appears to be something beyond our control. This same issue also happens with other plugins we tested, etc. Unfortunately we are unable to change this. 😕

**Reply by LauraMontgomery** [reply]
Hey @kjmartens thanks for investigating this. It sounds like something in the inner workings of Craft itself from what you are saying? Do you think this is something worth raising with Pixel and Tonic? It might be that they are unaware of this limitation around permissions.

**Comment by kjmartens** [comment]
Hi @LauraMontgomery,

We revisited this issue last week and we've added a new feature that makes this work. Once you update to Freeform 5.15.23+, check out the documentation here: 🙂
https://docs.solspace.com/craft/freeform/v5/configuration/permissions/#form-access-delegation

**Reply by LauraMontgomery** [reply]
Amazing, thank you so much for adding this :) I've mentioned internally about this update to Freeform and likely that we'll implement on the original project in the very near future - I reckon they'll very much still want to implement it
